src/plugin/plugin.cpp

```diff
@@ -58,56 +58,66 @@ namespace linuxdeploy {
+                            continue;
+                        }
 
-                    if (!std::regex_match(i->path().filename().string().c_str(), res, PLUGIN_EXPR)) {
-                        ldLog() << LD_DEBUG << "Doesn't match plugin regex, skipping:" << i->path() << std::endl;
-                        continue;
-                    }
+                        // entry name must match regular expression
+                        std::cmatch res;
 
-                    try {
-                        auto name = res[1].str();
-                        auto* plugin = createPluginInstance(*i);
-                        if (plugin == nullptr) {
-                            ldLog() << LD_DEBUG << "Failed to create instance for plugin" << i->path() << std::endl;;
-                        } else {
-                            ldLog() << LD_DEBUG << "Found plugin '" << LD_NO_SPACE << name << LD_NO_SPACE << "':" << plugin->path() << std::endl;
+                        if (!std::regex_match(i->path().filename().string().c_str(), res, PLUGIN_EXPR)) {
+                            ldLog() << LD_DEBUG << "Doesn't match plugin regex, skipping:" << i->path() << std::endl;
+                            continue;
+                        }
 
-                            if (foundPlugins.find(name) != foundPlugins.end()) {
-                                ldLog() << LD_DEBUG << "Already found" << name << "plugin in" << foundPlugins[name]->path() << std::endl;
+                        try {
+                            auto name = res[1].str();
+                            auto* plugin = createPluginInstance(*i);
+                            if (plugin == nullptr) {
+                                ldLog() << LD_DEBUG << "Failed to create instance for plugin" << i->path() << std::endl;;
                             } else {
-                                foundPlugins[name] = plugin;
+                                ldLog() << LD_DEBUG << "Found plugin '" << LD_NO_SPACE << name << LD_NO_SPACE << "':" << plugin->path() << std::endl;
+
+                                if (foundPlugins.find(name) != foundPlugins.end()) {
+                                    ldLog() << LD_DEBUG << "Already found" << name << "plugin in" << foundPlugins[name]->path() << std::endl;
+                                } else {
+                                    foundPlugins[name] = plugin;
+                                }
                             }
+                        } catch (const PluginError& e) {
+                            ldLog() << LD_WARNING << "Could not load plugin" << i->path() << LD_NO_SPACE << ": " << e.what();
                         }
-                    } catch (const PluginError& e) {
-                        ldLog() << LD_WARNING << "Could not load plugin" << i->path() << LD_NO_SPACE << ": " << e.what();
                     }
+                } catch(std::filesystem::filesystem_error const& ex) {
+                    ldLog() << LD_WARNING
```

Comment:
Should be `LD_DEBUG`, I guess. Please fix the indentation to match the other code.